server/channels/app/notification_push.go

```diff
@@ -138,6 +140,25 @@ func (a *App) sendPushNotificationToAllSessions(msg *model.PushNotification, use
 }
 
 func (a *App) sendPushNotification(notification *PostNotification, user *model.User, explicitMention, channelWideMention bool, replyToThreadType string) {
+	cancelled := false
+	a.ch.RunMultiHook(func(hooks plugin.Hooks) bool {
```

Comment:
Dare I ask: what's the performance overhead when calls has registered this hook? (It it was more akin to `NotificationHasBeenPushed`, it's probably a non-issue, but now Calls processing every notification might be a bottleneck?)

Reply:
I don't have benchmarks, unfortunately. Since Calls will be the first (only, for now) implementer, @streamer45 and I will do a load test before our implementation is merged, using https://github.com/mattermost/enterprise/blob/3ddee2048e4f8d985da57a3ae8f43b075419eef4/metrics/metrics.go#L137-L139
Ticket: https://mattermost.atlassian.net/browse/MM-54171